### PR TITLE
Timeout in instance stop

### PR DESCRIPTION
**Pull request by pierlauro**
Added ability to specify stop timeout (default 10 - as for singularity instance stop specification)

### CODE REVIEW

**Comment by vsoch** [comment]
okay, looks like this goes back to at least [3.1](https://sylabs.io/guides/3.1/user-guide/cli/singularity_instance_stop.html) (and there are no docs for 3.0) so the check for version seems sufficient to honor the variable. Should we perhaps instead of doing a default, only add the variable given the case that the user has defined it _and_ we have Singularity version 3? This would be akin to honoring the Singularity default (and how it might change over time) instead of hard coding it here. Other than that small change, if you can confirm this works for you, I'm okay with not adding a test for now because it seems like it would be challenging to easily do (unless you have an idea?)

File: spython/instance/cmd/stop.py [hunk 1]
```diff
@@ -8,13 +8,15 @@
 from spython.logger import bot
 
 
-def stop(self, name=None, sudo=False):
+def stop(self, name=None, sudo=False, timeout=10):
```
Comment (vsoch) [inline spython/instance/cmd/stop.py]: E.g., here we would set `timeout=None` and update the docstring to say default uses Singularity default (unset), and then for the if just do:

```python
if "version 3" in self.version() and timeout:
    subgroup = ["instance", "stop", "-t", str(timeout)]
```

But I'm curious and want to open this up for discussion - have you found in practice that the default singularity timeout is not sufficient for most cases? If this is the case, then we likely would help users by having the default timeout of 10 (and the current implementation is good). Let me know your thoughts!

**Comment by pierlauro** [comment]
> okay, looks like this goes back to at least [3.1](https://sylabs.io/guides/3.1/user-guide/cli/singularity_instance_stop.html) (and there are no docs for 3.0) so the check for version seems sufficient to honor the variable. Should we perhaps instead of doing a default, only add the variable given the case that the user has defined it _and_ we have Singularity version 3? This would be akin to honoring the Singularity default (and how it might change over time) instead of hard coding it here. Other than that small change, if you can confirm this works for you, I'm okay with not adding a test for now because it seems like it would be challenging to easily do (unless you have an idea?)

I totally agree on the proposed change, this way (1) older singularity versions would not be affected and (2) there will be no problem if singularity changes the default timeout.

File: spython/instance/cmd/stop.py (hunk 1 above)
Comment (pierlauro) [reply spython/instance/cmd/stop.py]: The timeout is very useful when running applications that are making a lot of I/O at stop time: a common use case can be an in-memory system (plenty of GBs in memory) that need to offload data at once a computation is finished (plenty of GBs to write) requiring in most cases more than 10 seconds

File: spython/instance/cmd/stop.py [hunk 2]
```diff
@@ -29,6 +31,9 @@ def stop(self, name=None, sudo=False):
     if "version 3" in self.version():
         subgroup = ["instance", "stop"]
 
+    if timeout:
```
Comment (vsoch) [inline spython/instance/cmd/stop.py]: okay, so if Singularity version 2 ran this command the user would get an error from Singularity that there is no timeout, do you think that's okay?

Comment (pierlauro) [reply spython/instance/cmd/stop.py]: Sorry, I had already commited before receiving your previous comment, this way should be ok

**Comment by pierlauro** [comment]
Regarding the testing, as we're just making a call to singularity stop (that is already working) with an additional argument, I think it is not worth instrumenting some complex to setup time-based test

**Comment by vsoch** [comment]
Agree! And your changes look good.

**Comment by vsoch** [comment]
https://pypi.org/project/spython/0.0.77/

**Pull request merged**